Errors in Plugin Build

**Description:**

When attempting to build the Strapi plugin "strapi-plugin-location" using the command `npm run build`, I encounter TypeScript errors that indicate 'strapi.db' is possibly undefined.

**Environment:**
- Node.js version: Tried 16 and 19.
- npm version: 9.2.0
- TypeScript version: 5.2.2

**Steps to Reproduce:**
1. Clone the repository.
2. Install dependencies using `npm install`.
3. Run the build command using `npm run build`.
4. Observe the TypeScript errors.

**Error Messages:**
server/bootstrap.ts:13:14 - error TS18048: 'strapi.db' is possibly 'undefined'. const db = strapi.db.connection;

**Troubleshooting Steps Taken:**
- I tried many ways to solve it by changing the node version and deleting the node_modules folder + package-lock.json everytime and run `npm install` but still the same issue.

![image](https://github.com/notum-cz/strapi-plugin-location/assets/55892173/9b701979-373d-41fa-8884-67f859906c38)

Please let me know if you need any further assistance or have any specific details to add to the issue description.

--- a/server/bootstrap.ts
+++ b/server/bootstrap.ts
@@ -9,6 +9,7 @@ export default async ({ strapi }: { strapi: Strapi }) => {
     // TODO: add information that plugin is disabled
     return;
   }
+  //@ts-expect-error
   const db = strapi.db.connection;
 
   const modelsWithLocation =
@@ -50,6 +51,7 @@ export default async ({ strapi }: { strapi: Strapi }) => {
   );
 
   const subscriber = createSubscriber(strapi);
+  //@ts-expect-error
   strapi.db.lifecycles.subscribe(subscriber);
 
   const middleware = createFilterMiddleware(strapi);

--- a/server/destroy.ts
+++ b/server/destroy.ts
@@ -1,6 +1,7 @@
 import { Strapi } from "@strapi/strapi";
 
 export default async ({ strapi }: { strapi: Strapi }) => {
+  //@ts-expect-error
   const db = strapi.db.connection;
 
   const modelsWithLocation =

--- a/server/register.ts
+++ b/server/register.ts
@@ -47,7 +47,6 @@ export default async ({ strapi }: { strapi: Strapi }) => {
 
   strapi.customFields.register({
     name: "location",
-    //@ts-expect-error
     plugin: pluginId,
     type: "json",
     inputSize: {

--- a/server/utils/lifecycles.ts
+++ b/server/utils/lifecycles.ts
@@ -5,6 +5,7 @@ import { Subscriber } from "@strapi/database/lib/lifecycles/subscribers";
 const locaitonServiceUid = "plugin::location-plugin.locationServices";
 
 const createSubscriber = (strapi: Strapi): Subscriber => {
+  //@ts-expect-error
   const db = strapi.db.connection;
   const modelsWithLocation =
     strapi.services[locaitonServiceUid].getModelsWithLocation();

--- a/server/utils/middleware.ts
+++ b/server/utils/middleware.ts
@@ -13,6 +13,7 @@ type LocationQueryCombined = LocationQuery | LogicalQuery;
 
 const locaitonServiceUid = "plugin::location-plugin.locationServices";
 const createFilterMiddleware = (strapi: Strapi) => {
+  //@ts-expect-error
   const db = strapi.db.connection;
   const modelsWithLocation =
     strapi.services[locaitonServiceUid].getModelsWithLocation();

